File: 617.MergeTwoBinaryTrees/Solution.py
class Solution:
    def mergeTrees(self, t1, t2):
        """
        :type t1: TreeNode
        :type t2: TreeNode
        :rtype: TreeNode
        """
        #set t1 as the main tree
        if t1 is not None and t2 is not None:
            #update t1
            #update value
            t1.val += t2.val
            #update left
            t1.left = self.mergeTrees(t1.left, t2.left)
            #update right
            t1.right = self.mergeTrees(t1.right, t2.right)
            
            return t1
        return t1 if t2 is None else t2

File: 617.MergeTwoBinaryTrees/Solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func mergeTrees(t1 *TreeNode, t2 *TreeNode) *TreeNode {
	if t1 != nil && t2 != nil {
		//create a new node
		val := t1.Val + t2.Val
		node := &TreeNode{Val: val}
		node.Left = mergeTrees(t1.Left, t2.Left)
		node.Right = mergeTrees(t1.Right, t2.Right)
		return node
	}
	if t1 != nil {
		return t1
	}
	return t2
}

































